### PR TITLE
Add long string support.

### PR DESCRIPTION
**Pull request by zherczeg**
An option for fixing the string issue: support strings with uint32 length. Not necessarily the best option.

### CODE REVIEW

**Comment by LaszloLango** [comment]
Is this related to #1274?

**Comment by zherczeg** [comment]
Kind of an alternative. Not sure this is the best approach, so this is just a proposal.

**Comment by zherczeg** [comment]
The performance change is just noise, but the binary size is increased by 600 bytes (156952->157512). Question is, do we want this?

File: jerry-core/ecma/base/ecma-globals.h [hunk 1]
```diff
@@ -872,32 +875,33 @@ typedef struct ecma_string_t
...
+    lit_magic_string_ex_id_t magic_string_ex_id; /**< identifier of an external magic string */
+    ecma_value_t lit_number; /**< literal number (note: not a regular string type) */
+    uint32_t common_field; /**< for zeroing and comparison in some cases */
   } u;
 } ecma_string_t;
 
 /**
+ * Long ECMA string-value descriptor
+ */
+typedef struct ecma_long_string_t
```
Comment (LaszloLango) [inline jerry-core/ecma/base/ecma-globals.h]: Please write only `typedef struct`

**Comment by LaszloLango** [comment]
I think we need this change. LGTM after the style fix.

**Comment by dbatyai** [comment]
This will surely be useful and in my opinion the trade-off is definitely worth it. LGTM

**Pull request merged**